StorageControl/src/V2/Client/StorageControlClient.php

```diff
@@ -0,0 +1,589 @@
+ *
+ * @method PromiseInterface createFolderAsync(CreateFolderRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface createManagedFolderAsync(CreateManagedFolderRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface deleteFolderAsync(DeleteFolderRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface deleteManagedFolderAsync(DeleteManagedFolderRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface getFolderAsync(GetFolderRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface getManagedFolderAsync(GetManagedFolderRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface getStorageLayoutAsync(GetStorageLayoutRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface listFoldersAsync(ListFoldersRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface listManagedFoldersAsync(ListManagedFoldersRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface renameFolderAsync(RenameFolderRequest $request, array $optionalArgs = [])
+ */
+final class StorageControlClient
+{
+    use GapicClientTrait;
+    use ResourceHelperTrait;
+
+    /** The name of the service. */
+    private const SERVICE_NAME = 'google.storage.control.v2.StorageControl';
+
+    /**
+     * The default address of the service.
+     *
+     * @deprecated SERVICE_ADDRESS_TEMPLATE should be used instead.
+     */
+    private const SERVICE_ADDRESS = 'storage.googleapis.com';
+
+    /** The address template of the service. */
+    private const SERVICE_ADDRESS_TEMPLATE = 'storage.UNIVERSE_DOMAIN';
+
+    /** The default port of the service. */
+    private const DEFAULT_SERVICE_PORT = 443;
+
+    /** The name of the code generator, to be included in the agent header. */
+    private const CODEGEN_NAME = 'gapic';
+
+    /** The default scopes required by the service. */
+    public static $serviceScopes = [
+        'https://www.googleapis.com/auth/cloud-platform',
+        'https://www.googleapis.com/auth/cloud-platform.read-only',
+        'https://www.googleapis.com/auth/devstorage.full_control',
+        'https://www.googleapis.com/auth/devstorage.read_only',
+        'https://www.googleapis.com/auth/devstorage.read_write',
+    ];
+
+    private $operationsClient;
+
+    private static function getClientDefaults()
+    {
+        return [
+            'serviceName' => self::SERVICE_NAME,
+            'apiEndpoint' => self::SERVICE_ADDRESS . ':' . self::DEFAULT_SERVICE_PORT,
+            'clientConfig' => __DIR__ . '/../resources/storage_control_client_config.json',
+            'descriptorsConfigPath' => __DIR__ . '/../resources/storage_control_descriptor_config.php',
+            'gcpApiConfigPath' => __DIR__ . '/../resources/storage_control_grpc_config.json',
+            'credentialsConfig' => [
+                'defaultScopes' => self::$serviceScopes,
+            ],
+            'transportConfig' => [
+                'rest' => [
```

Comment:
This mentions rest though.

Reply:
You're right! Thank you for pointing this out. We should be able to remove this in https://github.com/googleapis/gapic-generator-php/pull/707